### PR TITLE
chore: add SonarCloud scan + coverage analysis

### PR DESCRIPTION
**Pull request by KrisQ**
### WHY are these changes introduced?

Fixes [[ENG-7229](https://nacelle.atlassian.net/browse/ENG-7229)] 

### WHAT is this pull request doing?

- configure SonarCloud for shopify-cart

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -5,7 +5,6 @@ on:
   push:
     branches:
       - main
-      - ENG-6843/shopify-cart-1.0.0
```
Comment (NWRichmond) [inline .github/workflows/release.yml:8]: I don't think we can remove this until #278.

**Review by NWRichmond — CHANGES_REQUESTED** [review]
Looks good! The only issue I see is that we can't yet remove `ENG-6843/shopify-cart-1.0.0` as a release branch, as doing so would prevent us from releasing `@beta` versions of `@nacelle/shopify-cart` before #278 is merged.

File: .github/workflows/release.yml (hunk 1 above)
Comment (KrisQ) [reply .github/workflows/release.yml]: My bad, should be fixed now! Thank you @NWRichmond

**Review by NWRichmond — APPROVED** [review]
Looks good! Thanks for that update, @KrisQ.

**Pull request merged**